### DISCUSSION TITLE
Building an Operator

### DISCUSSION THREAD

**Discussion by vitorpmoreira** [kubernetes-client/csharp#1185] [Q&A]
Hello,

Does this client offer features to ease the implementation of an Operator? I don't mean a full SDK-style thing as the _Operator SDK_, just some layers of abstraction in order to transparently implement the controller pattern.
Or the idea is to use the API in a plain way and implement the algorithm, so-to-speak?

Thanks :)!

**Comment by tg123** [comment]
https://github.com/buehler/dotnet-operator-sdk

i know there are more similar sdks, but forget the links

**Reply by vitorpmoreira** [reply]
I've seen that one, I was just hoping that this client also provided some features since I'm already using it.
Nevertheless, thanks for your reply. I'm going to think about it. Maybe it's even ok to just use this client since what I need is for one pod (that uses this client and is authenticated in the cluster through its ServiceAccount) to have an internal storage of the existence of some other pods and, for that, having it run a watch of a Resource and/or CustomResource is more than enough.

**Comment by IvanJosipovic** [comment]
The YARP project is a good example and it includes an Informer, https://github.com/microsoft/reverse-proxy/tree/main/src/Kubernetes.Controller

**Reply by vitorpmoreira** [reply]
Wow, didn't know they were (also) making it an Ingress Controller. Thanks for sharing :)! That will certainly help!

**Reply by IvanJosipovic** [reply]
Here is the related issue https://github.com/microsoft/reverse-proxy/issues/1254
